cmd/concourse/quickstart.go

```diff
@@ -3,35 +3,133 @@ package main
 import (
 	"crypto/rand"
 	"crypto/rsa"
+	"errors"
 	"fmt"
 	"os"
 
+	"github.com/clarafu/envstruct"
+	"github.com/concourse/concourse/atc/atccmd"
 	concourseCmd "github.com/concourse/concourse/cmd"
+	v "github.com/concourse/concourse/cmd/concourse/validator"
+	"github.com/concourse/concourse/tsa/tsacmd"
 	"github.com/concourse/concourse/worker/workercmd"
 	"github.com/concourse/flag"
-	"github.com/jessevdk/go-flags"
+	"github.com/go-playground/locales/en"
+	ut "github.com/go-playground/universal-translator"
+	"github.com/go-playground/validator/v10"
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
 	"github.com/tedsuo/ifrit"
 	"github.com/tedsuo/ifrit/grouper"
 	"github.com/tedsuo/ifrit/sigmon"
 	"golang.org/x/crypto/ssh"
+	"gopkg.in/yaml.v2"
 )
 
-type QuickstartCommand struct {
-	*WebCommand              `group:"Web Configuration"`
-	*workercmd.WorkerCommand `group:"Worker Configuration" namespace:"worker"`
+var quickStart QuickstartConfig
+
+var QuickstartCommand = &cobra.Command{
+	Use:   "quickstart",
+	Short: "TODO",
+	Long: `Run both 'web' and 'worker' together, auto-wired.
+	Not recommended for productionTBD`,
+	RunE: InitializeQuickstart,
+}
+
+func init() {
+	QuickstartCommand.Flags().StringVar(&quickStart.ConfigFile, "config", "", "path to the config file that will be used to quickstart the concourse cluster")
+
+	quickStart.WebConfig.RunConfig = atccmd.CmdDefaults
+	quickStart.WebConfig.TSAConfig = tsacmd.CmdDefaults
+	quickStart.WorkerCommand = workercmd.CmdDefaults
+
+	// IMPORTANT!: Can be removed when flags no longer supported
+	atccmd.InitializeATCFlagsDEPRECATED(QuickstartCommand, &quickStart.WebConfig.RunConfig)
+	tsacmd.InitializeTSAFlagsDEPRECATED(QuickstartCommand, &quickStart.WebConfig.TSAConfig)
+	workercmd.InitializeWorkerFlagsDEPRECATED(QuickstartCommand, &quickStart.WorkerCommand, "worker-")
 }
 
-func (cmd QuickstartCommand) LessenRequirements(command *flags.Command) {
-	cmd.WebCommand.LessenRequirements(command)
-	cmd.WorkerCommand.LessenRequirements("worker-", command)
+type QuickstartConfig struct {
+	ConfigFile string `env:"QUICKSTART_CONFIG_FILE"`
+
+	WebConfig               `yaml:"web" ignore_env:"true"`
+	workercmd.WorkerCommand `yaml:"worker"`
+}
+
+func InitializeQuickstart(cmd *cobra.Command, args []string) error {
+	// IMPORTANT! This can be removed after we completely deprecate flags
+	fixupFlagDefaults(cmd, &quickStart.WebConfig)
+
+	// Fetch out env values
+	env := envstruct.Envstruct{
+		Prefix:        "CONCOURSE",
+		TagName:       "yaml",
+		OverrideName:  "env",
+		IgnoreTagName: "ignore_env",
+		StripValue:    true,
+
+		Parser: envstruct.Parser{
+			Delimiter:   ",",
```

Comment:
What's this do? (same reasoning as above) What does it delimit? hm, maybe slices and maps?

Reply:
Yup! Slices and maps